lib/spaceship/tunes/tester.rb

```diff
@@ -113,8 +119,12 @@ def all_by_app(app_id)
         # @param app_id (String) (required): The app id to filter the testers
```

Comment:
in this case I am trying to be very careful about backward compatibility. For this method it is in fact required, however if it is nil it will delegate as shown.  This is to allow consumers (pilot) to always use the *_by_app methods -- moving several null checks and branches into this class.  

There is a larger refactor of this particular class needed to really address some of the repitition and method overlap.  Ping me on Slack if you want to discuss further.